### DISCUSSION TITLE
Is it possible to convert pytorch to ggml?

### DISCUSSION THREAD

**Discussion by Dampish0** [ggml-org/ggml#65] [Q&A] [answered]
Is it possible to convert pytorch to ggml? and how would i do so?

**Comment by regularfry** [comment]
I'm interested in this too.  The Salesforce CodeGen model is published as a pytorch model and it's *awfully* close to gpt-j, which this can already handle.  Feels like we're one patch away from a fully local Copilot (among other, less obvious use cases).

**Comment by dchest** [comment, marked as answer]
There's a script to convert Cerebras pytorch model to ggml:
https://github.com/ggerganov/ggml/blob/master/examples/gpt-2/convert-cerebras-to-ggml.py

cformers repo have more converters, including codegen:
https://github.com/NolanoOrg/cformers/tree/master/cformers/cpp/converters

**Comment by narain1** [comment]
would it be nice if he have a general purpose layer translator [Yolo Model Regpresentation](https://github.com/ultralytics/yolov5/blob/master/models/yolov5l.yaml)
